### DISCUSSION TITLE
Backup of the volumes of finished job (pod in completed state)

### DISCUSSION THREAD

**Discussion by maratsal** [vmware-tanzu/velero#3469] [Community support Q&A]
I just was curious is there a way to backup the volume of finished cron job?

We had a cronjobs that are doing proper cold backup of our application and storing all data on PVC. I wanted to do a backup of those volumes by using restic backup, but seems pods have to be running to make it possible to back up the volumes.

Is there any solution for this?

For now I was thinking to mount those volumes to te running pods too :) 
or refactor whole backup solution and do only backup with velero by using pre-backup hook.

**Comment by oitTim** [comment]
We are noticing the same behavior.  Our cronjob PVC's are not getting backed up using Velero File System Backup.  The metadata of the cronjob is, however the PVC data is not being backed up.

Do you use FSB or do you use snapshots to do your backups?  We are wondering if we use snapshots, will it work then? Maybe the snapshots talk to the device instead of the host mount?

I see that this is a limitation listed here: https://velero.io/docs/main/file-system-backup/ 

Velero’s File System Backup reads/writes data from volumes by accessing the node’s filesystem, on which the pod is running. For this reason, FSB can only backup volumes that are mounted by a pod and not directly from the PVC. For orphan PVC/PV pairs (without running pods), some Velero users overcame this limitation running a staging pod (i.e. a busybox or alpine container with an infinite sleep) to mount these PVC/PV pairs prior taking a Velero backup.

**Reply by sseago** [reply]
@oitTim Filesystem backups require a running pod due to the way it is implemented. Without a running pod, the PVC is not mounted on any of the nodes, so the node agent doesn't have access to the filesystem. If you use snapshots, it should work fine. Note that if you're using snapshots+datamover, it will only work if your storageclass has the `Immediate` volume binding mode rather than `WaitForFirstConsumer` because with WFFC, the volume isn't bound to the PVC until there's a pod scheduled.

**Reply by oitTim** [reply]
@sseago Thanks for the reply. We will switch over to snapshots and see how it works. We are using dataMover, so will also look at changing the storageClass as well.

I appreciate the fast and informative response!
